hri_safety_sense/src/VscProcess.h

```diff
@@ -30,58 +28,67 @@
 #include "hri_c_driver/VehicleMessages.h"
 #include "hri_c_driver/VehicleInterface.h"
 
-using namespace hri_safety_sense_srvs;
+#include "hri_safety_sense_srvs/srv/emergency_stop.hpp"
+#include "hri_safety_sense_srvs/srv/key_value.hpp"
+#include "hri_safety_sense_srvs/srv/key_string.hpp"
 
 namespace hri_safety_sense {
 
-	// Diagnostics
-	struct ErrorCounterType {
-		uint32_t sendErrorCount;
-		uint32_t invalidRxMsgCount;
-	};
-
-	/**
-	 * Local Definitions
-	 */
-	const unsigned int VSC_INTERFACE_RATE = 50; /* 50 Hz */
-	const unsigned int VSC_HEARTBEAT_RATE = 20; /* 20 Hz */
-
-	class VscProcess {
-	   public:
-		  VscProcess();
-		  ~VscProcess();
-
-		  // Main loop
-		  void processOneLoop(const ros::TimerEvent&);
-
-		  // ROS Callback's
-		  bool EmergencyStop(EmergencyStop::Request &req, EmergencyStop::Response &res);
-		  bool KeyValue(KeyValue::Request &req, KeyValue::Response &res);
-		  bool KeyString(KeyString::Request &req, KeyString::Response &res);
-
-	   private:
-
-		  void readFromVehicle();
-		  int handleHeartbeatMsg(VscMsgType& recvMsg);
-
-		  // Local State
-		  uint32_t 				myEStopState;
-		  ErrorCounterType 		errorCounts;
-
-		  // ROS
-		  ros::NodeHandle 		rosNode;
-		  ros::Timer 	  		mainLoopTimer;
-		  ros::ServiceServer    estopServ, keyValueServ, keyStringServ;
-		  ros::Publisher		estopPub;
-		  ros::Time 			lastDataRx, lastTxTime;
-
-		  // Message Handlers
-		  MsgHandler			*joystickHandler;
-
-		  /* File descriptor for VSC Interface */
-		  VscInterfaceType		*vscInterface;
-
-	};
+  // Diagnostics
+  struct ErrorCounterType {
+    uint32_t sendErrorCount;
+    uint32_t invalidRxMsgCount;
+  };
+
+  /**
+   * Local Definitions
+   */
+  const unsigned int VSC_INTERFACE_RATE = 50; /* 50 Hz */
+  const unsigned int VSC_HEARTBEAT_RATE = 20; /* 20 Hz */
+
+  class VscProcess : public rclcpp::Node {
```

Comment:
You can make this class `final`, since nothing will be deriving from it.

Reply:
7de7162